rasa/core/policies/rule_policy.py

```diff
@@ -535,6 +556,9 @@ def _check_prediction(
             for rule_name, action_name in self._rules_sources[prediction_source]
             if action_name != gold_action_name
         }
+        if self._should_delete(prediction_source, tracker):
+            self.lookup[RULES].pop(prediction_source)
+            return []
```

Comment:
I'd move it to line `552`, to create a clear visual separation when there is no contradiction

Reply:
I don't think you moved it